app/react/Library/components/SortButtons.tsx

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import { bindActionCreators, Dispatch } from 'redux';
+import { connect, ConnectedProps } from 'react-redux';
+import { actions } from 'react-redux-form';
+import { t } from 'app/I18N';
+import { wrapDispatch } from 'app/Multireducer';
+import { Icon } from 'UI';
+import { DropdownList } from 'app/Forms';
+import { propertyTypes } from 'shared/propertyTypes';
+import { ObjectIdSchema, PropertySchema } from 'shared/types/commonTypes';
+import { IImmutable } from 'shared/types/Immutable';
+import { ClientTemplateSchema, IStore } from 'app/istore';
+
+const isSortableType = (type: PropertySchema['type']) => {
+  switch (type) {
+    case propertyTypes.text:
+    case propertyTypes.date:
+    case propertyTypes.numeric:
+    case propertyTypes.select:
+      return true;
+    default:
+      return false;
+  }
+};
+
+type SortType = {
+  label: string;
+  name: string;
+  value: string;
+  type: string;
+  context?: ObjectIdSchema;
+};
+
+type SearchOptions = {
+  order?: 'asc' | 'desc';
+  sort?: string;
+  searchTerm?: string;
+};
+
+const isSortable = (property: PropertySchema) =>
+  property.filter &&
+  (isSortableType(property.type) || (property.inherit && isSortableType(property.inherit.type!)));
+
+const getSortString = (property: PropertySchema) =>
+  `metadata.${property.name}${property.inherit ? '.inheritedValue' : ''}`;
+
+const getMetadataSorts = (templates: IImmutable<ClientTemplateSchema[]>) =>
+  templates.toJS().reduce((sorts: SortType[], template: ClientTemplateSchema) => {
+    (template.properties || []).forEach((property: PropertySchema) => {
+      if (isSortable(property) && !sorts.find(s => s.name === property.name)) {
+        sorts.push({
+          label: property.label,
+          name: property.name,
+          value: getSortString(property),
+          type: property.type,
+          context: template._id,
+        });
+      }
+    });
+    return sorts;
+  }, []);
+
+interface SortButtonsOwnProps {
+  stateProperty: string;
+  // eslint-disable-next-line react/no-unused-prop-types
```

Comment:
This can be avoided by returning from `mapStateToProps`all the `ownProps` that are destructured in the component definition, and removing `SortButtonsOwnProps` from the type definition in `mappedProps`. 
It's sort of centralizing the type and component props in the `mapStateToProps` function.
It makes no difference in terms of type definitions, so I don't know what approach is best from a code style point of view.